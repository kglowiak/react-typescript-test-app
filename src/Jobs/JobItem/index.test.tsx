import React from 'react';
import { render } from 'enzyme';

import { JobItemProps, JobItem } from './index';

describe('JobItem', () => {
  const props: JobItemProps = {
    job: {
      type: 'job',
      time: 123455,
      score: 1,
      by: 'test',
      id: 123455,
      title: 'test 1',
      url: 'testurl',
    },
  };

  it('should render a component', () => {
    const wrapper = render(<JobItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
