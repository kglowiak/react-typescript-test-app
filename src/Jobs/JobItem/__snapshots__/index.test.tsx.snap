// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem should render a component 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-10mb6uy",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-1m1p7ie",
          "href": "testurl",
          "target": "_blank",
        },
        "children": Array [
          Object {
            "data": "test 1",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-uahybi",
          },
          "children": Array [
            Object {
              "data": "Added by test at 02/01/1970",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "href": undefined,
          "target": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "href": undefined,
          "target": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-uahybi",
        },
        "children": Array [
          Object {
            "data": "Added by test at 02/01/1970",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-1m1p7ie",
            "href": "testurl",
            "target": "_blank",
          },
          "children": Array [
            Object {
              "data": "test 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "a",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "href": undefined,
            "target": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
